client/pages/projects/index.jsx

```diff
@@ -14,62 +14,276 @@ import {
+
+  const getSearchBarText = () => {
+    const text = document.getElementById("user-input").value;
+
+    return text;
+  };
+
+  const options = {
+    keys: ["name", "description"]
+  };
+
+  const populateAllProjects = async () => {
+    const response = await getModels(null, true);
+    setAllProjects(response.data);
+    setProjects(response.data);
+  };
 
   useEffect(() => {
-    const populateProjects = async () => {
-      const numProjects = 21;
+    populateAllProjects();
+  }, []);
+
+  useEffect(() => {
+    const populateDropDownFilteredProjects = () => {
+      let dropdownFilteredModels = [];
+
+      const models = allProjects;
+
+      let isMatchingSDGs = false;
+      let isMatchingCountry = false;
+      let isMatchingGrpSize = false;
+      let isMatchingDifficulty = false;
+
+      if (
+        selectedUNGoals === null &&
+        selectedCountry === null &&
+        selectedGrpSize === null &&
+        selectedDifficulty === null
+      ) {
+        return [];
+      }
+
+      let sdgSelectedNums = [];
+
+      if (selectedUNGoals !== null) {
+        for (let i = 0; i < selectedUNGoals.length; i++) {
+          sdgSelectedNums.push(parseInt(selectedUNGoals[i].value));
+        }
+      }
+
+      for (let i = 0; i < models.length; i++) {
+        if (selectedUNGoals === null) {
+          isMatchingSDGs = true;
+        } else if (
+          selectedUNGoals !== null &&
+          selectedUNGoals.length <= models[i].sdg.length
+        ) {
+          const matches = models[i].sdg.filter(sdg =>
+            sdgSelectedNums.includes(sdg)
+          );
+
+          isMatchingSDGs = matches.length === sdgSelectedNums.length;
+        }
+
+        if (
+          selectedCountry === null ||
+          models[i].country === selectedCountry.label
+        ) {
+          isMatchingCountry = true;
+        }
+
+        if (
+          selectedGrpSize === null ||
+          models[i].groupSize === selectedGrpSize.label
+        ) {
+          isMatchingGrpSize = true;
+        }
 
-      const models = await getModels();
-      setProjects(models.data.slice(numProjects));
+        if (
+          selectedDifficulty === null ||
+          models[i].difficulty === selectedDifficulty.label.toLowerCase()
+        ) {
+          isMatchingDifficulty = true;
+        }
+
+        if (
+          isMatchingSDGs &&
+          isMatchingCountry &&
+          isMatchingGrpSize &&
+          isMatchingDifficulty
+        ) {
+          dropdownFilteredModels.push(models[i]);
+        }
+
+        isMatchingSDGs = false;
+        isMatchingCountry = false;
+        isMatchingGrpSize = false;
+        isMatchingDifficulty = false;
+      }
+
+      if (dropdownFilteredModels.length === 0) {
+        setAlert(true);
+      }
+
+      return dropdownFilteredModels;
     };
 
-    populateProjects();
-  }, []);
+    const populateSearchFilteredProjects = () => {
+      let searchFilteredModels = [];
+
+      let textInput = getSearchBarText();
```

Comment:
this line (and `getSearchBarText`) can be replaced with `userInput`, as the values will always be the same